### PR TITLE
add wave v0.5.1 (Wave Terminal)

### PR DESCRIPTION
**Pull request by sawka**
- [x] The submission is for a [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online waveterm` is error-free.
- [x] `brew style --fix waveterm` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

Wave Terminal is a new Open Source (Apache 2.0) terminal, available for MacOS and Linux (cask covers only the MacOS build).  We just soft launched the terminal over Thanksgiving and have already had a couple thousand downloads, and over 700 Github Stars.  Wave is maintained by Command Line Inc and as of next week we'll have 5 full-time engineers working on, maintaining, and improving the application.  Our universal build has been signed and notarized.  I'm the founder/creator of Wave Terminal (so I'm authorized to upload/maintain this cask), and am happy to answer any questions or make any changes to the cask that might be required.

Note the application is installed as "Wave.app", but I chose the token "waveterm" to match with our website (https://waveterm.dev), our Github repo (https://github.com/wavetermdev/waveterm), and to clearly show that it is a command line terminal.  We are planning on using the token "waveterm" for other package managers as well (snap, flatpak, aur, etc.).

### CODE REVIEW

File: Casks/w/waveterm.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  homepage "https://www.waveterm.dev/"
+
+  livecheck do
+    url "https://github.com/wavetermdev/waveterm"
+    strategy :github_releases
+  end
+
+  auto_updates false
+  depends_on macos: ">= :catalina"
+  depends_on arch: [:intel, :arm64]
```
Comment (bevanjkay) [inline Casks/w/waveterm.rb:18]: This line can be omitted as this is the default.

File: Casks/w/waveterm.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  name "WaveTerm"
+  desc "Modern Terminal for Seamless Workflows"
+  homepage "https://www.waveterm.dev/"
+
+  livecheck do
+    url "https://github.com/wavetermdev/waveterm"
+    strategy :github_releases
+  end
+
+  auto_updates false
```
Comment (bevanjkay) [inline Casks/w/waveterm.rb:16]: This line can be omitted as it is the default

File: Casks/w/waveterm.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+cask "waveterm" do
```
Comment (bevanjkay) [inline Casks/w/waveterm.rb:1]: The cask `token` should be based upon the `app` stanza, in this case "wave'

File: Casks/w/waveterm.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  url "https://dl.waveterm.dev/builds/waveterm-macos-universal-v#{version}.dmg"
+  name "Wave Terminal"
+  name "WaveTerm"
+  desc "Modern Terminal for Seamless Workflows"
+  homepage "https://www.waveterm.dev/"
+
+  livecheck do
+    url "https://github.com/wavetermdev/waveterm"
+    strategy :github_releases
+  end
```
Comment (bevanjkay) [inline Casks/w/waveterm.rb:14]:
```suggestion
  livecheck do
    url "https://www.waveterm.dev/download"
    regex(/href=.*?waveterm.+[._-]v?(\d+(?:\.\d+)+)\.dmg/i)
  end
```

For the `livecheck` we should follow the source used for the `url`.

**Review by bevanjkay — CHANGES_REQUESTED** [review]

File: Casks/w/waveterm.rb (hunk 1 above)
Comment (sawka) [reply Casks/w/waveterm.rb:18]: sounds good, will remove

File: Casks/w/waveterm.rb (hunk 2 above)
Comment (sawka) [reply Casks/w/waveterm.rb:16]: sounds good, will remove

File: Casks/w/waveterm.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+cask "waveterm" do
+  version "0.5.1"
+  sha256 "fe965fe581a66bbd7c394dbfd7ebb1635cd0bf46329042ef86db123fbd7d5751"
+
+  url "https://dl.waveterm.dev/builds/waveterm-macos-universal-v#{version}.dmg"
+  name "Wave Terminal"
+  name "WaveTerm"
+  desc "Modern Terminal for Seamless Workflows"
```
Comment (bevanjkay) [inline Casks/w/waveterm.rb:8]:
```suggestion
  desc "Terminal emulator"
```

You are welcome to suggest an alternative, but it should adhere to the `[guidelines](https://docs.brew.sh/Cask-Cookbook#stanza-desc)`.

File: Casks/w/waveterm.rb (hunk 4 above)
Comment (sawka) [reply Casks/w/waveterm.rb:14]: ok, happy to change to the download page for livecheck.  just tested the regexp and updated it to this which now passes the audit:
`regex(/href=.*?dl\.waveterm\.dev.+-v(\d+(?:\.\d+)+)\.dmg/i)`

File: Casks/w/waveterm.rb (hunk 5 above)
Comment (sawka) [reply Casks/w/waveterm.rb:8]: got it.  missed those guidelines when writing the desc, will update.

File: Casks/w/waveterm.rb (hunk 3 above)
Comment (sawka) [reply Casks/w/waveterm.rb:1]: Hi Bevan, I was hoping to keep the token "waveterm" as that is the token we are using throughout all of our branding, including our website, socials, github, and for other package managers (in the future).  if this is a hard requirement that you cannot make an exception for, lmk and I'll change the token.

File: Casks/w/waveterm.rb (hunk 4 above)
Comment (bevanjkay) [reply Casks/w/waveterm.rb]: Sorry, I missed part of the regex, I'll push a fix now.

File: Casks/w/waveterm.rb (hunk 3 above)
Comment (bevanjkay) [reply Casks/w/waveterm.rb:1]: @sawka `token` is a hard requirement to ensure consistency. https://docs.brew.sh/Cask-Cookbook#token-reference

File: Casks/w/waveterm.rb (hunk 4 above)
Comment (bevanjkay) [reply Casks/w/waveterm.rb]: I don't have permission to push to the branch, so here's the suggestion.

```suggestion
  livecheck do
    url "https://www.waveterm.dev/download"
    regex(/href=.*?waveterm.+?[._-]v?(\d+(?:\.\d+)+)\.dmg/i)
  end
```

File: Casks/w/waveterm.rb (hunk 3 above)
Comment (sawka) [reply Casks/w/waveterm.rb:1]: ok, will update

File: Casks/w/waveterm.rb (hunk 4 above)
Comment (bevanjkay) [reply Casks/w/waveterm.rb]: No worries. Can you please just push this final suggestion as it aligns with our standard regex patterns more closely, and then we should be good to go.

File: Casks/w/wave.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  url "https://dl.waveterm.dev/builds/waveterm-macos-universal-v#{version}.dmg"
+  name "Wave Terminal"
+  name "WaveTerm"
+  desc "Terminal emulator"
+  homepage "https://www.waveterm.dev/"
+
+  livecheck do
+    url "https://www.waveterm.dev/download"
+    regex(/href=.*?dl\.waveterm\.dev.+-v(\d+(?:\.\d+)+)\.dmg/i)
+  end
```
Comment (bevanjkay) [inline Casks/w/wave.rb:14]:
```suggestion
  livecheck do
    url "https://www.waveterm.dev/download"
    regex(/href=.*?waveterm.+?[._-]v?(\d+(?:\.\d+)+)\.dmg/i)
  end
```

This regex matches our standard patterns better.

File: Casks/w/wave.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+  depends_on macos: ">= :catalina"
+
+  app "Wave.app"
+
+  zap trash: [
+    "~/.waveterm",
+  ]
```
Comment (bevanjkay) [inline Casks/w/wave.rb:22]:
```suggestion
  zap trash "~/.waveterm"
```

**Review by bevanjkay — CHANGES_REQUESTED** [review]
Two final changes and we should be good to go.

File: Casks/w/wave.rb (hunk 6 above)
Comment (sawka) [reply Casks/w/wave.rb]: updated

File: Casks/w/wave.rb (hunk 7 above)
Comment (sawka) [reply Casks/w/wave.rb]: done (just had to add a colon after trash, so ended up with `zap trash: "~/.waveterm"`)

**Comment by bevanjkay** [comment]
Thank you @sawka for your quick responses, and sorry for all the back and forth

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**